Doc/c-api/hash.rst

```diff
@@ -41,6 +45,29 @@ See also the :c:member:`PyTypeObject.tp_hash` member.
    .. versionadded:: 3.4
 
 
+Functions
+^^^^^^^^^
+
+.. c:function:: Py_hash_t Py_HashDouble(double value)
+
+   Hash a C double number.
+
+   * If *value* is positive infinity, return :data:`sys.hash_info.inf
+     <sys.hash_info>`.
+   * If *value* is negative infinity, return :data:`-sys.hash_info.inf
+     <sys.hash_info>`.
+   * If *value* is not-a-number (NaN), return :data:`sys.hash_info.nan
+     <sys.hash_info>` (``0``).
+   * Otherwise, return the hash value of the finite *value* number.
+
+   .. note::
+      Return the hash value ``0`` for the floating point numbers ``-0.0`` and
+      ``+0.0``, and for not-a-number (NaN). ``Py_IS_NAN(value)`` can be used to
+      check if *value* is not-a-number.
```

Comment:
It exposes too much implementation details why already exposed in different place. Why not simply say that it is equivalent to hash() of Python float object if it is not a NaN? And if it is a NaN, you should use other value to avoid collisions.

Reply:
Are you talking about the note, or describing the 3 cases and return values? I can just remove the note. My idea is to suggest using Py_IS_NAN() to treate NaN differently. But I'm not sure which implementation to suggest.

@zooba says that if you have a Python object, just call `PyObject_Hash(obj)` on it 😁

Reply:
About describing all 3 cases. It should already be described in other place (documentation for `sys.hash_info` or `float` or `hash()`), and if it is not described in details, than it is not necessary for users. You should only document that for non-NaN values it returns the same result as for hash() for Python float object.